internal/services/redisenterprise/redis_enterprise_database_data_source.go

```diff
@@ -25,6 +26,22 @@ func dataSourceRedisEnterpriseDatabase() *pluginsdk.Resource {
 			Required:     true,
 			ValidateFunc: redisenterprise.ValidateRedisEnterpriseID,
 		},
+		// TODO: deprecate me
+		"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
```

Comment:
can we deprecate this instead of adding a note?

forthe deprecation we should be wrapping the property in the 4.0 flag so its removed when enabled (i think it exists now) and adding a depreaction message

Reply:
removed